tests/test_api.py

```diff
@@ -17,22 +16,19 @@ class AirbnbTestCase(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        """setup the class for testing"""
-        print("flask test")
-        cls.__engine = create_engine(
-            "mysql+mysqldb://{}:{}@{}/{}".format(
-                os.getenv("HBNB_MYSQL_USER"),
-                os.getenv("HBNB_MYSQL_PWD"),
-                os.getenv("HBNB_MYSQL_HOST"),
-                os.getenv("HBNB_MYSQL_DB")
-            ),
-            pool_pre_ping=True
-        )
-        Base.metadata.create_all(cls.__engine)
-        # with cls.__engine.connect() as con:
-        # with open("100-dump.sql", encoding='utf-8') as f:
-        #   dump = text(f.read())
-        # con.execute()
+        """setup the class for testing if storage type is db
+
+        """
+        if os.getenv("HBNB_TYPE_STORAGE") == "db"\
+                and os.getenv("HBNB_ENV") == "test":
+            with open("tests/100-dump.sql", "r") as f:
+                cmd = ["mysql", "-uroot", "-pdocker"]
+                proc = subprocess.Popen(
+                    cmd,
+                    stdin=f,
+                    stderr=subprocess.DEVNULL
+                )
+                err, out = proc.communicate()
         cls.__app = app.test_client()
 
```

Comment:
I believe a teardown to clear out the engine created here would be safer across the other unites classes.